### PR TITLE
Fix: Env variabled to pick up right simulator device for tests

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
We are seeing this error when we run e2e tests 
image-20241121-053448

Available devices on thie github runner image 
image-20241121-053421

JIRA: https://mattermost.atlassian.net/browse/MM-61905

```release-note
NONE
```

### CODE REVIEW

File: detox/.detoxrc.json [hunk 1]
```diff
@@ -30,8 +30,8 @@
     "ios.simulator": {
       "type": "ios.simulator",
       "device": {
-        "type": "__DEVICE_TYPE__",
-        "os": "__DEVICE_OS__"
```
Comment (larkox) [inline detox/.detoxrc.json:34]: Why the name changes?

File: .github/workflows/e2e-detox-template.yml [hunk 2]
```diff
@@ -66,8 +66,8 @@ env:
   ADMIN_EMAIL: ${{ secrets.MM_MOBILE_E2E_ADMIN_EMAIL }}
   ADMIN_USERNAME: ${{ secrets.MM_MOBILE_E2E_ADMIN_USERNAME }}
   ADMIN_PASSWORD: ${{ secrets.MM_MOBILE_E2E_ADMIN_PASSWORD }}
-  # BRANCH: ${{ github.head_ref || github.ref_name }} 
-  # COMMIT_HASH: ${{ github.sha }}
```
Comment (larkox) [inline .github/workflows/e2e-detox-template.yml:70]: Is this intended?

**Review by larkox — APPROVED** [review]
Just a couple of questions

File: detox/.detoxrc.json (hunk 1 above)
Comment (yasserfaraazkhan) [reply detox/.detoxrc.json:34]: @larkox In the github actions, I have named it DEVICE_NAME for other workflows to pass any Device name and override. Wanted to keep local and ci with same reference to not get confused.

File: .github/workflows/e2e-detox-template.yml (hunk 2 above)
Comment (yasserfaraazkhan) [reply .github/workflows/e2e-detox-template.yml:70]: @larkox 
In reports, it always says HEAD. I'm trying to get the branch name to populate there. I see this change is not working.

I'll try some other options.
image

**Comment by yasserfaraazkhan** [comment]
Back to 82% pass rate https://community.mattermost.com/core/pl/5554i3hjnifkxqdc9j5unbscco

**Pull request merged**